File: plugins/bridge/android/src/main/kotlin/com/gyo/bridge/BridgeInterface.kt
package com.gyo.bridge

import org.json.JSONObject

/**
 * Interface for all Gyo bridge implementations
 */
interface BridgeInterface {
    /**
     * Unique name of the bridge (e.g., "gyo-camera")
     */
    val name: String
    
    /**
     * Invoke a bridge method
     * @param method Method name to invoke
     * @param data Optional data payload
     * @param callback Callback to return result or error
     */
    fun invoke(method: String, data: JSONObject?, callback: BridgeCallback)
}

/**
 * Callback interface for bridge methods
 */
interface BridgeCallback {
    /**
     * Called when the bridge method succeeds
     * @param data Result data (can be String, JSONObject, JSONArray, etc.)
     */
    fun resolve(data: Any?)
    
    /**
     * Called when the bridge method fails
     * @param error Error message
     */
    fun reject(error: String)
}

File: plugins/bridge/ios/Sources/GyoBridge/BridgeInterface.swift
import Foundation

/**
 * Result type for bridge callbacks
 */
public enum BridgeResult {
    case success(Any?)
    case failure(String)
}

/**
 * Callback type for bridge methods
 */
public typealias BridgeCallback = (BridgeResult) -> Void

/**
 * Protocol for all Gyo bridge implementations
 */
public protocol BridgeInterface {
    /**
     * Unique name of the bridge (e.g., "gyo-camera")
     */
    var name: String { get }
    
    /**
     * Invoke a bridge method
     * - Parameters:
     *   - method: Method name to invoke
     *   - data: Optional data payload as dictionary
     *   - callback: Callback to return result or error
     */
    func invoke(method: String, data: [String: Any]?, callback: @escaping BridgeCallback)
}
